### PR TITLE
Update narayana to 5.12.4 and exclude some resteasy bits

### PR DESCRIPTION
**Pull request by famod**
Also follows up on #20390.

/cc @mmusgrov

PS: Apologies if this is premature. I just subscribed for relases after #20390, saw 5.12.4 (and that there hasn't been an update for a while) and simply gave it a try.

### CODE REVIEW

File: bom/application/pom.xml [hunk 1]
```diff
@@ -4186,6 +4186,14 @@
...
                         <artifactId>commons-logging</artifactId>
                     </exclusion>
+                    <exclusion>
+                        <groupId>org.jboss.resteasy</groupId>
+                        <artifactId>resteasy-client</artifactId>
+                    </exclusion>
+                    <exclusion>
+                        <groupId>org.jboss.resteasy</groupId>
+                        <artifactId>resteasy-jackson2-provider</artifactId>
+                    </exclusion>
```
Comment (famod) [inline bom/application/pom.xml:4196]: These two resteasy dependencies were added in 5.12.1:
https://mvnrepository.com/artifact/org.jboss.narayana.rts/lra-client/5.12.0.Final
https://mvnrepository.com/artifact/org.jboss.narayana.rts/lra-client/5.12.1.Final
(and were updated in 5.12.2)

As far as I understood #21172, users shall have the choice which resteasy flavor they prefer, so excluding classic resteasy made sense to me. Also, without these exclusions there will be a banned dependency to jakarta-activation.
/cc @geoand

Comment (geoand) [reply bom/application/pom.xml:4196]: Right, that is what https://github.com/quarkusio/quarkus/pull/21172 does

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
@famod is the documentation clear about the fact that the RESTEasy bits need to be added?

**Comment by famod** [comment]
@gsmet
It would say so, yes: https://github.com/quarkusio/quarkus/blob/main/docs/src/main/asciidoc/lra.adoc#configuration
@geoand took care of that in #21172 back then.

**Pull request merged**